### DISCUSSION TITLE
Make design primitives React-independent

### DISCUSSION THREAD

**Discussion by thomasheyenbrock** [graphql/graphiql#2585] [Ideas]
The current "reference implementation" of GraphiQL is using React. However, there are a lot of other frameworks and libraries that would benefit from having a GraphQL IDE that can be easily embedded.

A first step towards this could be to extract our design primitives (styles) out of `@graphiql/react` into a different packages, maybe `@graphiql/toolkit` or maybe a new dedicated package.

There's not yet any research or alignment of how this would look like and work exactly in code. If you stumble upon this discussion, feel free to drop your needs / ideas / feedback!

**Comment by acao** [comment]
@thomasheyenbrock a separate package sounds like a great idea as well! `@graphiql/design` for example, that can be imported by `@graphiql/react` and/or `graphiql`

**Comment by jonathanawesome** [comment]
Pulling design tokens out into a separate package is a great idea and the perfect first step to modularizing the component structure. A typed theme object that, at first, includes a basic set of tokens that could be easily consumable across frameworks may even bring additional folks into the community.

Another idea would be to gather community consensus and publish an svg/icon package that sets a base for visually describing the GraphQL spec. I've been particularly gushy over [this bit](https://www.figma.com/file/p8f8MaG7ZRnuCc4lVv7w6t/GraphiQL-2.0?node-id=1891%3A35587) in the GraphiQL 2.0 design file. Screenshot here for reference:
<img width="229" alt="Screen Shot 2022-07-26 at 7 47 27 PM" src="https://user-images.githubusercontent.com/963528/181139817-d46a33c4-fa62-4c6b-ad0e-6cedeca01dde.png">

If we're having a conversation about the reference implementation becoming agnostic, I think the visual language is a wonderful place to start.
